src/vcpkg/configuration.cpp

```diff
@@ -571,6 +571,28 @@ namespace vcpkg
         }
     }
 
+    static bool registry_config_requests_configure_environment(const Optional<RegistryConfig>& target) noexcept
```

Comment:
```suggestion
    static bool registry_config_requests_configure_environment(Optional<const RegistryConfig&> target) noexcept
```
I think the `any_of` below is copying the entire configuration. If this change doesn't compile, then I'd like to see instead something like:

```cpp
static bool registry_config_requests_configure_environment(const RegistryConfig& target) noexcept;

bool Configuration::requests_configure_environment() const
{
    if (!ce_metadata.is_empty()) return true;
    if (auto reg = default_reg.get()) {
        if (registry_config_requests_configure_environment(*reg)) return true;
    }
    return std::any_of(registries.begin(), registries.end(), registry_config_requests_configure_environment);
}
```

Reply:
Ouch, maybe that ctor of Optional should be explicit

Reply:
I went with this second one because `Optional<T&>` is evil and particularly evil as a parameter.